File: solutions/go/isbn-verifier/1/isbn_verifier.go
// Package isbn provides a function for checking isbn.
package isbn

import (
	"strings"
	"unicode"
)

// IsValidISBN returns whether or not the given isbn-10 is valid.
func IsValidISBN(isbn string) bool {
	mult := 10
	sum := 0

	if strings.HasSuffix(isbn, "X") {
		isbn = strings.Replace(isbn, "X", "0", -1)
		sum += 10
	}

	if len(isbn) < 10 || !unicode.IsDigit(rune(isbn[len(isbn)-1])) {
		return false
	}

	for _, char := range isbn {
		if unicode.IsDigit(char) {
			sum += mult * int(char-'0')
			mult -= 1
		} else if char != '-' {
			return false
		}
		if mult < 0 {
			return false
		}
	}

	return sum%11 == 0
}

File: solutions/python/isbn-verifier/1/isbn_verifier.py
def is_valid(isbn):
    """
    Checks the validity of an ISBN-10 number.

    Args:
        isbn (str): The ISBN-10 number to be validated.

    Returns:
        bool: True if `isbn` is a valid ISBN-10, False otherwise.
    """
    new_isbn = "".join(c for c in isbn if c.isdigit() or c == "X")

    if len(new_isbn) != 10 or any(c not in "0123456789X-" for c in isbn):
        return False

    return (
        sum(10 if c == "X" else int(c) * i for c, i in zip(new_isbn, range(10, 0, -1)))
        % 11
        == 0
    )
